src/Jackalope/Node.php

```diff
@@ -819,6 +825,42 @@ public function getPrimaryItem()
     }
 
     /**
+     * Set the UUID generator to use. If not set, the phpcr-utils UUIDHelper
+     * will be used.
+     *
+     * @param callable $generator
+     */
+    public function setUuidGenerator(\Closure $generator)
+    {
+        $this->uuidGenerator = $generator;
+    }
+
+    /**
+     * @return callable a uuid generator function.
+     */
+    protected function getUuidGenerator()
```

Comment:
why do we need the additional flexibility with the uuid generator? and is this the best way we can do it? and now that https://github.com/phpcr/phpcr-utils/pull/159 is merged, do we need this?

i would see this more as a 2.0 topic, to make the UuidHelper non-static but an interface that we inject or get from the factory or however we build these things in 2.0.

would it be enough to drop the special case in the ValueConverter and leave this change out?